src/init.cpp

```diff
@@ -1761,7 +1761,11 @@ bool AppInitMain(InitInterfaces& interfaces)
     connOptions.nLocalServices = nLocalServices;
     connOptions.nMaxConnections = nMaxConnections;
     connOptions.m_max_outbound_full_relay = std::min(MAX_OUTBOUND_FULL_RELAY_CONNECTIONS, connOptions.nMaxConnections);
-    connOptions.m_max_outbound_block_relay = std::min(MAX_BLOCKS_ONLY_CONNECTIONS, connOptions.nMaxConnections-connOptions.m_max_outbound_full_relay);
+    if (IsReachable(NET_ONION) && (IsReachable(NET_IPV4) || IsReachable(NET_IPV6))) {
```

Comment:
Not sure how all this tor stuff works, but I was only able to get this code to trigger by explicitly setting `-onion=<>` as part of my bitcoind config, even though I *think* bitcoind will listen for inbound over tor by default if it can connect to the control port even without that command-line option?  Does that sound right?

Reply:
@sdaftuar if Tor is correctly configured on your machine and if `listen` isn't 0 then you should see a hidden service announced in your debug log, and it should be listening on that.
https://github.com/bitcoin/bitcoin/blob/master/doc/tor.md#3-automatically-listen-on-tor

Reply:
TorControl is started as a thread by default a few lines above this one, and sets `NET_ONION` to be "reachable" if it starts up correctly:
https://github.com/bitcoin/bitcoin/blob/65d12110d43ac89bc0320477bfea6b375fc2134b/src/torcontrol.cpp#L525-L531

Which means that if `NET_ONION` is deemed not "reachable" before TorControl is started, then the behavior here would not be deterministic. As in, the `IsReachable(NET_ONION)` here (here=this line in this PR we're reviewing) would depend on whether the `TorControl` thread has exectued far enough to set `NET_ONION` as "reachable".